superset/assets/javascripts/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -14,6 +14,18 @@ class QueryAutoRefresh extends React.PureComponent {
   componentWillUnmount() {
     this.stopTimer();
   }
+  shouldCheckForQueries() {
+    // if there are started or pending queries, this method should return true
+    const { queries } = this.props;
+    const queryKeys = Object.keys(queries);
+    /* eslint consistent-return: 0 */
+    const pendingOrStartedQueries = queryKeys.forEach((key) => {
+      if (queries[key].state === 'running' || queries[key].state === 'started') {
+        return true;
+      }
+    });
+    return pendingOrStartedQueries;
```

Comment:
I guess it works because forEach returns null or undefined, which evals to false